rest/utilities_testing_user.go

```diff
@@ -37,6 +37,7 @@ func MakeUser(t *testing.T, httpClient *http.Client, serverURL, username, passwo
 		if err != nil {
 			return true, err, resp
 		}
+		assert.NoError(t, resp.Body.Close())
```

Comment:
Is it possible for the code outside the retry loop to read the body after it has already been closed? (47-51)

I think we can also drop the outer Body.Close() if we ensure the unclosed body doesn't leak out of the retry loop.